src/sync/deployment.ts

```diff
@@ -49,12 +72,39 @@ const getTransformedDeployments = async (deployments, gitHubInstallationClient:
 	});
 
 	const transformedDeployments = await Promise.all(transformTasks);
-	return transformedDeployments
-		.filter(deployment => !!deployment)
+	return (transformedDeployments
+		.filter(deployment => !!deployment) as JiraDeploymentBulkSubmitData[])
 		.map(deployment => deployment.deployments)
 		.flat();
 };
 
+const saveDeploymentsForLaterUse = async (deployments: FetchDeploymentResponse["deployments"], gitHubBaseUrl: string, logger: Logger) => {
+	try {
+		const successDeployments: FetchDeploymentResponse["deployments"] = deployments.filter(d => (d.statuses?.nodes.some(n => n.state === "SUCCESS")));
+		logger.info({ deploymentsCount: deployments.length, successDeploymentsCount: successDeployments.length }, "Try to save deployments for later use");
+		const result = await Promise.allSettled(successDeployments.map(dep => {
+			const successStatusDate = dep.statuses?.nodes.find(n=>n.state === "SUCCESS")?.updatedAt;
+			if (!successStatusDate) {
+				logger.warn("Should find a success status date, but found none");
+				throw "Save failure";
```

Comment:
Can we change this to throw new Error("Save failure")? It's generally better to throw an instance of the Error object rather than throwing a simple string as the Error object provides more info about the error. This will provide more context about what went wrong. Also, the error object will capture the stack trace and we can leverage it for better error handling.

Reply:
Good call!!
Fixed.